[core] Allows to add custom export item (#3891)
mui/mui-x: docs/data/data-grid/export/export.md

For more option to customize the print export, please visit the [`printOptions` api page](/api/data-grid/grid-print-export-options/).

> ## Custom export

Reviewer: ```suggestion
## Custom export format
```

You can add custom export formats by creating your own export menu.
To simplify its creation, we export `<GridToolbarExportContainer />` which contains the menu logic.
The default `<GridToolbarExport />` is defined as follow:

```jsx
const GridToolbarExport = ({ csvOptions, printOptions, ...other }) => (
  <GridToolbarExportContainer {...other}>
    <GridCsvExportMenuItem options={csvOptions} />
    <GridPrintExportMenuItem options={printOptions} />
  </GridToolbarExportContainer>
);
```

Each child of the `<GridToolbarExportContainer />` receives a prop `hideMenu` to close the export menu after the export.

> Here is an example of how to add a JSON export.

Reviewer: ```suggestion
The demo below shows how to add a JSON export.
```
